src/intervals/bed.jl

```diff
@@ -73,6 +73,17 @@ end
 "An `Interval` with associated metadata from a BED file"
 typealias BEDInterval Interval{BEDMetadata}
 
+function Base.open(filepath::AbstractString, mode::AbstractString, ::Type{BED};
+                   n_fields::Integer=-1)
+    io = open(filepath, mode)
+    if mode[1] == 'r'
+        return open(BufferedInputStream(io), BED)
+    elseif mode[1] ∈ ('w', 'o')
+        return BEDWriter(io, n_fields)
+    end
+    error("invalid open mode")
```

Comment:
Yes, but it is unavoidable unless we unify the parser and the writer types into a type.